ptvsd/wrapper.py

```diff
@@ -1674,23 +1695,22 @@ def _new_sock():
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     return sock
 
+def _add_socket_handler(client, pydevd, name, killonclose=True, addhandlers=True):
+    proc = VSCodeMessageProcessor(client, pydevd, killonclose=killonclose)
 
-def _start(client, server, killonclose=True, addhandlers=True):
-    name = 'ptvsd.Client' if server is None else 'ptvsd.Server'
-
-    pydevd = PydevdSocket(lambda *args: proc.on_pydevd_event(*args))
-    proc = VSCodeMessageProcessor(client, pydevd,
-                                  killonclose=killonclose)
-
-    server_thread = threading.Thread(target=proc.process_messages,
-                                     name=name)
+    server_thread = threading.Thread(target=proc.process_messages, name=name)
     server_thread.daemon = True
     server_thread.start()
 
     if addhandlers:
         _add_atexit_handler(proc, server_thread)
         _set_signal_handlers(proc)
```

Comment:
I'm going to have to modify `_set_signal_handlers` to close all VS Code message processors.